commands/create-account.js

```diff
@@ -39,6 +41,34 @@ module.exports = {
 async function createAccount(options) {
     await eventtracking.track(eventtracking.EVENT_ID_CREATE_ACCOUNT_START, { nodeUrl: options.nodeUrl });
     // NOTE: initialBalance is passed as part of config here, parsed in middleware/initial-balance
+    // periods are disallowed in top-level accounts and can only be used for subaccounts
+    const splitAccount = options.accountId.split('.');
+
+    const splitMaster = options.masterAccount.split('.');
+    const masterRootTLA = splitMaster[splitMaster.length - 1];
+    if (splitAccount.length === 1) {
+        // TLA (bob-with-at-least-maximum-characters)
+        if (splitAccount[0].length < TLA_MIN_LENGTH) {
+            console.log(`Top-level accounts must be greater than ${TLA_MIN_LENGTH} characters.\n` +
+              'Note: this is for advanced usage only. Typical account names are of the form:\n' +
+              'app.alice.test, where the masterAccount shares the top-level account (.test).'
+            );
+            return;
+        }
+    } else if (splitAccount.length > 1) {
+        // Subaccounts (short.alice.near, even.more.bob.test, and eventually peter.potato)
+        // Check that master account TLA matches
+        const accountTLA = splitAccount.filter((n, i) => i !== 0).join('.');
+        if (accountTLA !== options.masterAccount) {
+            console.log(`New account doesn't share the same top-level account. Expecting account name to end in ".${options.masterAccount}"`);
+            return;
+        }
+
+        // Warn user if account seems to be using wrong network, where TLA is captured in config
```

Comment:
nit: `Object.prototype.hasOwnProperty.call(options, 'tla')` -> `options.tla` ? Do not think want to treat `tla: null` or `tla: ''` as `tla` being present.